crates/stackable-versioned-macros/src/codegen/vstruct/mod.rs

```diff
@@ -319,22 +319,40 @@ impl VersionedStruct {
         crd_fn_calls: Vec<TokenStream>,
         enum_variants: Vec<(Ident, String)>,
     ) -> TokenStream {
-        let ident = &self.idents.kubernetes;
+        let enum_ident = &self.idents.kubernetes;
+        let enum_vis = &self.visibility;
 
-        let version_enum_definition = self.generate_kubernetes_version_enum(enum_variants);
+        let mut enum_variant_matches = TokenStream::new();
+        let mut enum_variant_idents = TokenStream::new();
+
+        for (enum_variant_ident, enum_variant_display) in enum_variants {
+            enum_variant_idents.extend(quote! {#enum_variant_ident,});
+            enum_variant_matches.extend(quote! {
+                #enum_ident::#enum_variant_ident => f.write_str(#enum_variant_display),
+            });
+        }
 
         quote! {
             #[automatically_derived]
-            pub struct #ident;
+            #enum_vis enum #enum_ident {
+                #enum_variant_idents
+            }
```

Comment:
**thought (non-blocker)**: I think ultimately this will move outside of the kubernetes specific stuff. Eg: for semver, there might be `Foo:1_2_3`. But for now, it is fine here.

Reply:
Maybe the versions should be available even if the k8s feature is disabled.

Reply:
Yes, I can see that it might be useful to generate the versions even when no K8s specific features are used. Do we want to make the required changes in this PR?

Reply:
I'd be ok moving it later.